### PR TITLE
metastore: init at 1.1.2

### PR DESCRIPTION
**Pull request by haskelious**
###### Motivation for this change
Small Linux specific tool that is very useful to store metadata information of a folder structure (metadata such as file owners, groups, permissions) into a metadata "backup" file. Can then be used to restore/reapply the permissions later.

The typical use case is to store metadata of a folder structure in a git repository (along with the file changes) so a checkout of git can reapply the metadata intended by the original commit.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -6790,6 +6790,12 @@
     githubId = 4061736;
     name = "Severen Redwood";
   };
+  sstef = {
+    email = "stephane@nix.frozenid.net";
+    github = "fkstef";
+    githubId = 8668915;
+    name = "Stephane Schitter";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: should be a separate commit
```
maintainers: add sstef
```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    owner = "przemoc";
+    repo = "metastore";
+    rev = "v${version}";
+    sha256 = "0mb10wfckswqgi0bq25ncgabnd3iwj7s7hhg3wpcyfgckdynwizv";
+  };
+
+  buildInputs = [ libbsd ];
+  installFlags = [ "DESTDIR=$(out)" "PREFIX=/" ];
+
+  meta = {
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]:
```suggestion
  meta = with stdenv.lib; {
```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    description = "Store and restore metadata from a filesystem";
+    homepage = "https://software.przemoc.net/#metastore";
+    license = stdenv.lib.licenses.gpl2;
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]:
```suggestion
    license = licenses.gpl2;
```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    maintainers = stdenv.lib.maintainers.sstef;
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]: needs to be an array
```suggestion
    maintainers = with maintainers; [ sstef ];
```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    platforms = stdenv.lib.platforms.linux;
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]:
```suggestion
    platforms = platforms.linux;
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (fkstef) [reply maintainers/maintainer-list.nix]: I moved this part to this pull request: https://github.com/NixOS/nixpkgs/pull/80954

**Comment by fkstef** [comment]
Ooops, something went terribly wrong with me trying to figure out how to update and rebase the commit !  I am not used to this yet. I will have to close this PR unless someone can help in getting it fixed.

**Comment by fkstef** [comment]
@jonringer would you have any advice how to clean up my mistake with my attempt to rebase, which apparently brought lots of unrelated commits to this pull request? Alternatively I can close this one to start from a new branch and pull request.

**Comment by jonringer** [comment]
assuming to just want the last 3 commits:

```
git rebase HEAD~3 --onto master
git push .. .. --force
```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{ stdenv, libbsd, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  version = "1.1.2";
+  pname = "metastore";
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]: please add new line between pname and src.
```suggestion
  pname = "metastore";

```

File: pkgs/os-specific/linux/metastore/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+  src = fetchFromGitHub {
+    owner = "przemoc";
+    repo = "metastore";
+    rev = "v${version}";
+    sha256 = "0mb10wfckswqgi0bq25ncgabnd3iwj7s7hhg3wpcyfgckdynwizv";
+  };
+
+  buildInputs = [ libbsd ];
+  installFlags = [ "DESTDIR=$(out)" "PREFIX=/" ];
```
Comment (jonringer) [inline pkgs/os-specific/linux/metastore/default.nix]:
```suggestion
  installFlags = [ "PREFIX=$(out)" ];
```

**Review by jonringer — COMMENTED** [review]
otherwise LGTM

**Comment by fkstef** [comment]
I believe I start to understand how to build these PR. Thank you for your support @jonringer.

File: pkgs/os-specific/linux/metastore/default.nix (hunk 6 above)
Comment (fkstef) [reply pkgs/os-specific/linux/metastore/default.nix]: I made the change, but your comment seems to stay there regardless

**Comment by jonringer** [comment]
each language framework/build tool chain is different.

**Comment by jonringer** [comment]
@GrahamcOfBorg build metastore

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM

```
[1 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/80914
1 package built:
metastore
```
